### PR TITLE
SONAR-7183 faster ce submit

### PR DESCRIPTION
**Pull request by sns-seb**
allow possibility to batch insert tasks into the CeQueue:
* using a single transaction for inserting all tasks
* using a single request to retrieve all components referenced by the inserted tasks rather than one per task

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/computation/queue/CeQueueImpl.java [hunk 1]
```diff
@@ -253,4 +287,69 @@ public void resumePeek() {
...
+      this.componentDtoByUuid = Collections.emptyMap();
+    }
+
+    public CeQueueDtoToCeTask(Map<String, ComponentDto> componentDtoByUuid) {
+      this.componentDtoByUuid = componentDtoByUuid;
+    }
+
+    @Override
+    @Nonnull
+    public CeTask apply(@Nonnull CeQueueDto dto) {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/computation/queue/CeQueueImpl.java]: MAJOR The "dto" parameter nullability is different in the superclass method, and that should not be changed. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2638)
MAJOR The return value nullability of this method is different in the superclass, and that should not be changed. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2638)

File: server/sonar-server/src/main/java/org/sonar/server/computation/queue/CeQueueImpl.java [hunk 2]
```diff
@@ -253,4 +287,69 @@ public void resumePeek() {
...
+      return dto;
+    }
+  }
+
+  private enum CeQueueDtoToComponentUuid implements Function<CeQueueDto, String> {
+    INSTANCE;
+
+    @Override
+    @Nullable
+    public String apply(@Nonnull CeQueueDto input) {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/computation/queue/CeQueueImpl.java]: MAJOR The "input" parameter nullability is different in the superclass method, and that should not be changed. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2638)

File: server/sonar-server/src/main/java/org/sonar/server/computation/queue/CeQueueImpl.java [hunk 3]
```diff
@@ -253,4 +287,69 @@ public void resumePeek() {
...
+    private final DbClient dbClient;
+
+    public CeTaskSubmitToInsertedCeQueueDto(DbSession dbSession, DbClient dbClient) {
+      this.dbSession = dbSession;
+      this.dbClient = dbClient;
+    }
+
+    @Override
+    @Nonnull
+    public CeQueueDto apply(@Nonnull CeTaskSubmit submission) {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/computation/queue/CeQueueImpl.java]: MAJOR The "submission" parameter nullability is different in the superclass method, and that should not be changed. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2638)
MAJOR The return value nullability of this method is different in the superclass, and that should not be changed. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2638)

**Comment by SonarTech** [comment]
SonarQube analysis reported 6 issues:
* MAJOR 6 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR Add a 'package-info.java' file to document the 'protobuf' package (org.sonarsource.sonarqube:sonar-db:target/generated-sources/protobuf/org/sonar/db/protobuf) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)

**Comment by julienlancelot** [comment]
Add unit test(s) on CeQueue.massSubmit() then LGTM

**Comment by sns-seb** [comment]
unit tests added and many corner case tests added to existing code

**Comment by SonarTech** [comment]
SonarQube analysis reported 6 issues:
* MAJOR 6 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR Add a 'package-info.java' file to document the 'protobuf' package (org.sonarsource.sonarqube:sonar-db:target/generated-sources/protobuf/org/sonar/db/protobuf) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)
